src/test/components/elements/cards/EventCard.test.tsx

```diff
@@ -133,4 +135,56 @@ describe("EventCard", () => {
     const viewDetailsLink = viewDetails.querySelector("a");
     expect(viewDetailsLink).toHaveAttribute("href", `/events/${mockEvent.id}`);
   });
+
```

Comment:
can we be confident that the render() has completed before these not-exist tests run? There's the risk of a false pass if the render hasn't finished and we're only testing that something doesn't exist

Reply:
Is that necessary here? In this component all the data comes in as props.

Feels like it might be overkill to await the event title for example; there's no API requests

Reply:
```
const title = await screen.findByTestId("event-card-title");
expect(title).toBeInTheDocument();
```

The above code has been added to the tests to ensure that the title is rendered.
We can now be certain that the render() has been completed before the Join event now button is looked for.